tfhe/src/core_crypto/gpu/mod.rs

```diff
@@ -88,6 +103,14 @@ impl CudaStreams {
     }
 }
 
+impl Clone for CudaStreams {
+    fn clone(&self) -> Self {
+        // The `new_multi_gpu_with_indexes()` function is used here to adapt to any specific type of
+        // streams being cloned (single, multi, or custom)
+        Self::new_multi_gpu_with_indexes(self.gpu_indexes.as_slice())
```

Comment:
We should not create new streams when calling clone, it should copy the streams instead (which is fine as they're pointers)

Reply:
No you cant just clone pointers, as in the Drop impl you destroy the streams,

So if Clone copy pointers you'll have some double free issues, unless you also change the drop impl

Reply:
Ok I was wrong indeed